registry/purview-registry/script/migrate_legacy_project.py

```diff
@@ -0,0 +1,179 @@
+from pyapacheatlas.core import (AtlasEntity, AtlasProcess, AtlasException,
+                                PurviewClient)
+from pyapacheatlas.core.util import GuidTracker
+
+'''
+To run this script, change default_purview_name, and run.
+Then the script will be able to add all necessary ProcessEntity for all projects.
+'''
+
+
+Label_Contains = "CONTAINS"
+Label_BelongsTo = "BELONGSTO"
+Label_Consumes = "CONSUMES"
+Label_Produces = "PRODUCES"
+guid = GuidTracker(starting=-1000)
+
+
+def _generate_relation_pairs(from_entity, to_entity, relation_type):
+    '''
+    This will be used to generate process entities between two entities
+    Since we are using bi-directional relation, we have pairs of (contain, belongsto) (consumes,produces) pairs
+    '''
+    global guid
+    type_lookup = {Label_Contains: Label_BelongsTo,
+                   Label_Consumes: Label_Produces}
+
+    # Process entity will have qualified name formatted as "TYPE__FromId__ToId"
+    # This means FromId has relation with TYPE to ToId. For example "CONTAINS__A__B" means A contains B, then "BELONSTO__B__A" should also be generated.
+    forward_relation = AtlasProcess(
+        name=str(from_entity["guid"]) + " to " + str(to_entity["guid"]),
+        typeName="Process",
+        qualified_name='__'.join(
+            [relation_type, str(from_entity["guid"]), str(to_entity["guid"])]),
+        inputs=[from_entity],
+        outputs=[to_entity],
+        guid=guid.get_guid())
+
+    backward_relation = AtlasProcess(
+        name=str(to_entity["guid"]) + " to " + str(from_entity["guid"]),
+        typeName="Process",
+        qualified_name='__'.join(
+            [type_lookup[relation_type], str(to_entity["guid"]), str(from_entity["guid"])]),
+        inputs=[to_entity],
+        outputs=[from_entity],
+        guid=guid.get_guid())
+    return [forward_relation, backward_relation]
+
+
+def upload_single_entity_to_purview(purview_client, entity: Union[AtlasEntity, AtlasProcess]):
+    ''' 
+    uplaod a single entity to purview, could be either AtlasEntity or AtlasProcess, return entity GUID.
+    If the entity with same qualified name already exists, PreConditionCheckFailed will be returned from Purview. 
+    In this case we will just use the existing entity.
+    '''
+    try:
+        entity.lastModifiedTS = "0"
+        result = purview_client.upload_entities([entity])
+        entity.guid = result['guidAssignments'][entity.guid]
+        print(
+            f"Successfully created {entity.typeName} -- {entity.qualifiedName}")
+    except AtlasException as e:
+        if "PreConditionCheckFailed" in e.args[0]:
+            entity.guid = purview_client.get_entity(
+                qualifiedName=entity.qualifiedName, typeName=entity.typeName)['entities'][0]['guid']
+            print(
+                f"Found existing entity  {entity.guid}, {entity.typeName} -- {entity.qualifiedName}")
+    return UUID(entity.guid)
+
+
+default_purview_name = "feathrazuretest3-purview1"
```

Comment:
move this to the top of the script so it's easier for users to find this

Reply:
Ok